### ISSUE TITLE
cannot work on windows && raspberry pi 3b

### ISSUE THREAD

**Issue by sndnvaps** [ajvb/kala#208]
i have run kala with cmd
```bash
kala run --port="8091"
```
then create new job with
```bash
$ curl -H "'Content-Type':'application/json'" -d '{"schedule":"R/2020-01-31T23:05:35/PT5M","name":"start_ebookdownloader_job","command":"/home/pi/gowork/src/github.com/sndnvaps/ebookdownloader/ebookdownloader_cli --bookid=91_91911 --txt --mobi"}' -X POST http://192.168.13.100:8091/api/v1/job/
```
got the crash 
```bash
ERRO[0454] Error occured when unmarshalling data: invalid character 's' looking for beginning of object key string 
[negroni] PANIC: runtime error: invalid memory address or nil pointer dereference
goroutine 100 [running]:
github.com/codegangsta/negroni.(*Recovery).ServeHTTP.func1(0x6632f048, 0x24cf9c0, 0x24d6460)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/recovery.go:34 +0x9c
panic(0x4c22a0, 0x8e5480)
        /usr/local/go/src/runtime/panic.go:679 +0x194
runtime/internal/atomic.goXadd64(0x24d8654, 0x1, 0x0, 0x24c6d01, 0xe5)
        /usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
github.com/cornelk/hashmap.(*List).insertAt(0x24d8650, 0x24cfc80, 0x0, 0x0, 0x0)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/list.go:129 +0x7c
github.com/cornelk/hashmap.(*List).AddOrUpdate(0x24d8650, 0x24cfc80, 0x0, 0x24c6d60)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/list.go:65 +0x118
github.com/cornelk/hashmap.(*HashMap).insertListElement(0x24d8660, 0x24cfc80, 0xa5072301, 0x73aec6a8)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/hashmap.go:152 +0x64
github.com/cornelk/hashmap.(*HashMap).Set(0x24d8660, 0x4a60c0, 0x24d2690, 0x257a580)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/hashmap.go:131 +0xa4
github.com/ajvb/kala/job.(*LockFreeJobCache).Set(0x2538140, 0x257a580, 0x24, 0x661800e0)
        /home/pi/gowork/src/github.com/ajvb/kala/job/cache.go:258 +0x50
github.com/ajvb/kala/job.(*Job).Init(0x257a580, 0x5e4dc0, 0x2538140, 0x0, 0x0)
        /home/pi/gowork/src/github.com/ajvb/kala/job/job.go:192 +0x184
github.com/ajvb/kala/api.HandleAddJob.func1(0x6632f048, 0x24cf9c0, 0x25b6000)
        /home/pi/gowork/src/github.com/ajvb/kala/api/api.go:139 +0xa4
net/http.HandlerFunc.ServeHTTP(0x24ce300, 0x6632f048, 0x24cf9c0, 0x25b6000)
        /usr/local/go/src/net/http/server.go:2007 +0x34
github.com/gorilla/mux.(*Router).ServeHTTP(0x24d0150, 0x6632f048, 0x24cf9c0, 0x25b6000)
        /home/pi/gowork/pkg/mod/github.com/gorilla/mux@v0.0.0-20150808061613-ffb3f683aad4/mux.go:100 +0x23c
github.com/codegangsta/negroni.Wrap.func1(0x6632f048, 0x24cf9c0, 0x25b6000, 0x24d6550)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/negroni.go:41 +0x3c
github.com/codegangsta/negroni.HandlerFunc.ServeHTTP(0x24d64b0, 0x6632f048, 0x24cf9c0, 0x25b6000, 0x24d6550)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/negroni.go:24 +0x3c
github.com/codegangsta/negroni.middleware.ServeHTTP(0x5e04f8, 0x24d64b0, 0x24d64e0, 0x6632f048, 0x24cf9c0, 0x25b6000)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/negroni.go:33 +0x84
github.com/ajvb/kala/api/middleware.(*Logger).ServeHTTP(0x255db00, 0x6632f048, 0x24cf9c0, 0x25b6000, 0x24d6540)
        /home/pi/gowork/src/github.com/ajvb/kala/api/middleware/logger.go:21 +0xe0
github.com/codegangsta/negroni.middleware.ServeHTTP(0x5df958, 0x255db00, 0x24d64d0, 0x6632f048, 0x24cf9c0, 0x25b6000)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/negroni.go:33 +0x84
github.com/codegangsta/negroni.(*Recovery).ServeHTTP(0x24d6460, 0x6632f048, 0x24cf9c0, 0x25b6000, 0x24d6530)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/recovery.go:45 +0x70
github.com/codegangsta/negroni.middleware.ServeHTTP(0x5df988, 0x24d6460, 0x24d64c0, 0x6632f048, 0x24cf9c0, 0x25b6000)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/negroni.go:33 +0x84
github.com/codegangsta/negroni.(*Negroni).ServeHTTP(0x24ced60, 0x5e3688, 0x2552120, 0x25b6000)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/negroni@v0.1.1-0.20150319171304-c7477ad8e330/negroni.go:73 +0xc4
net/http.serverHandler.ServeHTTP(0x2552090, 0x5e3688, 0x2552120, 0x25b6000)
        /usr/local/go/src/net/http/server.go:2802 +0x88
net/http.(*conn).serve(0x248c4e0, 0x5e4148, 0x25a4480)
        /usr/local/go/src/net/http/server.go:1890 +0x7e0
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2927 +0x2f0
```

**Comment by sndnvaps** [comment]
my go version:
```bash
go1.13.1
```

**Comment by sndnvaps** [comment]
when i restart the kala got some error 
```bash
1:6379
INFO[0000] Preparing cache                              
INFO[0000] Job test_job:9d3ec76d-1c58-421a-4cd3-5a12d4834228 repeating in 0s 
INFO[0000] Job test_job:9d3ec76d-1c58-421a-4cd3-5a12d4834228 added to cache. 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12cbc]

goroutine 1 [running]:
runtime/internal/atomic.goXadd64(0x240e6c4, 0x1, 0x0, 0x240d001, 0xc5)
        /usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
github.com/cornelk/hashmap.(*List).insertAt(0x240e6c0, 0x240d880, 0x0, 0x0, 0x0)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/list.go:129 +0x7c
github.com/cornelk/hashmap.(*List).AddOrUpdate(0x240e6c0, 0x240d880, 0x0, 0x240d0c0)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/list.go:65 +0x118
github.com/cornelk/hashmap.(*HashMap).insertListElement(0x240e6d0, 0x240d880, 0x3956ac01, 0x7cc3581b)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/hashmap.go:152 +0x64
github.com/cornelk/hashmap.(*HashMap).Set(0x240e6d0, 0x4a60c0, 0x2410bf8, 0x254c000)
        /home/pi/gowork/pkg/mod/github.com/cornelk/hashmap@v0.0.0-20170323055448-5fee90babde7/hashmap.go:131 +0xa4
github.com/ajvb/kala/job.(*LockFreeJobCache).Set(...)
        /home/pi/gowork/src/github.com/ajvb/kala/job/cache.go:258
github.com/ajvb/kala/job.(*LockFreeJobCache).Start(0x2540140, 0x2a05f200, 0x1, 0x7b8a800, 0xfffffff2)
        /home/pi/gowork/src/github.com/ajvb/kala/job/cache.go:202 +0x1dc
github.com/ajvb/kala/cmd.init.0.func2(0x24e8100)
        /home/pi/gowork/src/github.com/ajvb/kala/cmd/app.go:182 +0x408
github.com/codegangsta/cli.Command.Run(0x521bbf, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x523a0c, 0x8, 0x0, ...)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/cli@v1.8.1-0.20150711215404-bca61c476e3c/command.go:127 +0x3b4
github.com/codegangsta/cli.(*App).Run(0x251c360, 0x2484040, 0x4, 0x4, 0x0, 0x0)
        /home/pi/gowork/pkg/mod/github.com/codegangsta/cli@v1.8.1-0.20150711215404-bca61c476e3c/app.go:159 +0x3c0
main.main()
        /home/pi/gowork/src/github.com/ajvb/kala/main.go:20 +0x74
```

**Comment by tooolbox** [comment]
Seems caused by https://github.com/cornelk/hashmap

Reviewing the go.mod file, we are pinned to a version from 2017.  There have been ~50 commits since that point.

**Comment by tooolbox** [comment]
@sndnvaps how do you feel about trying with https://github.com/tooolbox/kala/tree/hashmap-update ?

**Comment by sndnvaps** [comment]
> @sndnvaps how do you feel about trying with https://github.com/tooolbox/kala/tree/hashmap-update ?

ok, i will try it now. thanks.

**Comment by sndnvaps** [comment]
@tooolbox i have test it . every thing is ok. 
i hope it can merge into master branch

**Comment by tooolbox** [comment]
Yup, merged & tagged as v0.3.3
